hypothesis-python/tests/nocover/test_exceptiongroup.py

```diff
@@ -0,0 +1,215 @@
+    # multiple UnsatisfiedAssumption => first one is reraised => engine suppresses it
+
+    @given(st.integers(min_value=0, max_value=3))
+    def test_function(val: int) -> None:
+        async def task(value: int) -> None:
+            if value == 0:
+                reject()
+
+        async def _main(value: int) -> None:
+            async with asyncio.TaskGroup() as tg:
+                tg.create_task(task(value))
+                tg.create_task(task(value))
+
+        asyncio.run(_main(val))
+
+    test_function()
+
+
+def test_exceptiongroup_multiple_InvalidArgument() -> None:
+    # multiple InvalidArgument => only first one is reraised... which seems bad.
+    # But raising a group might break ghostwriter(?)
+
+    @given(st.data())
+    def test_function(data: DataObject) -> None:
+        async def task1(d: DataObject) -> None:
+            d.draw(st.integers(max_value=1, min_value=3))
+
+        async def task2(d: DataObject) -> None:
+            d.draw(st.integers(max_value=2, min_value=3))
+
+        async def _main(d: DataObject) -> None:
+            async with asyncio.TaskGroup() as tg:
+                tg.create_task(task1(d))
+                tg.create_task(task2(d))
+
+        asyncio.run(_main(data))
+
+    with pytest.raises(errors.InvalidArgument):
+        test_function()
+
+
+# if intended, will probably remove this test, and either way probably belong somewhere else
+# data.conjecture_data is internal API, so calling freeze like this is invalid use of
+# internal API. I'm not sure if it's possible to repro this only using public API, in
+# which case it's much less of a problem - but could probably hide internal bugs at least.
+def test_frozen_things():
+    # Hypothesis reraises the TypeError as a StopTest, because the data is Frozen.
+    # Doesn't seem great, but I suppose it is intentional?
+    @given(st.data())
+    def foo(data):
+        data.conjecture_data.freeze()
+        raise TypeError("oops")
+
+    foo()
+
+
+# if above is intended, then this is supposedly also intended?
+def test_frozen_data_and_critical_user_exception():
+    @given(st.data())
+    def test_function(data):
+        data.conjecture_data.freeze()
+
+        async def task(d: DataObject) -> None:
+            d.draw(st.booleans())
+
+        async def task2() -> None:
+            raise TypeError("Critical error")
+
+        async def _main(d: DataObject) -> None:
+            async with asyncio.TaskGroup() as tg:
+                tg.create_task(task(d))
+                tg.create_task(task2())
+
+        asyncio.run(_main(data))
+
+    # does not raise anything, the TypeError is suppressed
+    test_function()
+    # with pytest.raises(ExceptionGroup) as exc_info:
+    # e = exc_info.value
+    # assert isinstance(e, ExceptionGroup)
+    # assert len(e.exceptions) == 2
+    # assert isinstance(e.exceptions[0], errors.Frozen)
+    # assert isinstance(e.exceptions[1], TypeError)
```

Comment:
this perhaps stands out as something to clean up/move elsewhere before merge, but otherwise I think it's mergeable

Reply:
I've just dug through, and I'm pretty sure it's impossible to hit this with public API - we only freeze the conjecturedata object after the user's code is done.  On that basis I'd be happy to just delete these tests and merge 🙂